compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/model/BIRNode.java

```diff
@@ -74,7 +74,7 @@ public BIRPackage(Location pos, Name org, Name pkgName, Name name, Name version,
                           String sourceRoot, boolean skipTest, boolean isTestPkg) {
             super(pos);
             packageID = new PackageID(org, pkgName, name, version, sourceFileName, sourceRoot, isTestPkg, skipTest);
-            this.importModules = new ArrayList<>();
+            this.importModules = new HashSet<>();
```

Comment:
Previously this is defined as a array, that means the logic may be depend on the order.
So shall we use a linked hashset or verify the order is not matter in the logic?

Reply:
Yes, this is a good suggestion. We use LinkedHashSet as a practice.

Reply:
Looked into the performance implications and it seems like both are almost similar. Since the `Set` in question is holding module imports, it is safe to assume it's size won't grow to a performance critical amount. 

Therefore, changed the type to a `LinkedHashSet` in b35b99e6205bbd6c9b8e2c393213506eca496959.